src/main/java/de/tum/in/www1/artemis/web/rest/lecture/OnlineUnitResource.java

```diff
@@ -0,0 +1,203 @@
+            throw new ConflictException("Lecture unit must be associated to a lecture of a course", "onlineUnit", "lectureOrCourseMissing");
+        }
+
+        validateUrl(onlineUnit);
+
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, onlineUnit.getLecture().getCourse(), null);
+
+        if (!onlineUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException("Requested lecture unit is not part of the specified lecture", "onlineUnit", "lectureIdMismatch");
+        }
+
+        OnlineUnit result = onlineUnitRepository.save(onlineUnit);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, onlineUnit.getId().toString())).body(result);
+    }
+
+    /**
+     * POST /lectures/:lectureId/online-units : creates a new online unit.
+     *
+     * @param lectureId      the id of the lecture to which the online unit should be added
+     * @param onlineUnit the online unit that should be created
+     * @return the ResponseEntity with status 201 (Created) and with body the new online unit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/lectures/{lectureId}/online-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<OnlineUnit> createOnlineUnit(@PathVariable Long lectureId, @RequestBody final OnlineUnit onlineUnit) throws URISyntaxException {
+        log.debug("REST request to create onlineUnit : {}", onlineUnit);
+        if (onlineUnit.getId() != null) {
+            throw new BadRequestException();
+        }
+
+        validateUrl(onlineUnit);
+
+        Lecture lecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoalsElseThrow(lectureId);
+        if (lecture.getCourse() == null) {
+            throw new ConflictException("Specified lecture is not part of a course", "onlineUnit", "courseMissing");
+        }
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, lecture.getCourse(), null);
+
+        // persist lecture unit before lecture to prevent "null index column for collection" error
+        onlineUnit.setLecture(null);
+        OnlineUnit persistedOnlineUnit = onlineUnitRepository.saveAndFlush(onlineUnit);
+
+        persistedOnlineUnit.setLecture(lecture);
+        lecture.addLectureUnit(persistedOnlineUnit);
+        lectureRepository.save(lecture);
+
+        return ResponseEntity.created(new URI("/api/online-units/" + persistedOnlineUnit.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, "")).body(persistedOnlineUnit);
+    }
+
+    /**
+     * Fetch the website's metadata from the specified link to an online resource
+     *
+     * @param link The link (as request parameter) to the website to fetch the metadata from
+     * @return A DTO with link, meta title, and meta description
+     */
+    @GetMapping("/online-resource")
```

Comment:
can we prefix the REST resource somehow?
And is it safe to connect to arbitrary URLs and load the whole document? this could be really large just to get some meta tags. Is this really worth the effort?

Reply:
one solution might be to only load the index.html, but preventing any subsequent loads of images, JavaScript files or additional requests to external pages

Reply:
another question which I couldn't answer by reading the PR description: are those values used to pre-fill the create dialog? (Then I would consider it as very useful)

Reply:
Resolved via Slack.